### ISSUE TITLE
Session state not filtering attributes (with EAP)

### ISSUE THREAD

**Issue by johnnywalker** [FreeRADIUS/freeradius-server#893]
When using the new `outer.session-state` configuration instead of `use_tunneled_reply`, attributes aren't getting filtered at the end of the EAP sequence (*tunnel success > Access-Challenge > Access-Request > Access-Accept*).

Here's where they are getting cached after the *tunnel success* and before the final *Access-Challenge*:
```
session-state: Saving cached attributes
  MS-MPPE-Encryption-Policy += Encryption-Allowed
  MS-MPPE-Encryption-Types += RC4-40or128-bit-Allowed
  MS-MPPE-Send-Key += 0x41bec7335e750efe3e5d889041e406c5
  MS-MPPE-Recv-Key += 0xca6711a13fe9ebd182791f155b9d9e3f
  EAP-Message += 0x030c0004
  Message-Authenticator += 0x00000000000000000000000000000000
Sent Access-Challenge Id 62 from 10.x.x.x:1812 to 10.x.x.x:1645 length 101
  EAP-Message = 0x010d002b19001703010020f3f6ec51e9f807f6e307716856b4b1bad7352016ce6f285ad651bcdb9ab907e7
  Message-Authenticator = 0x00000000000000000000000000000000
  State = 0xe6d18ccaeddc95a5a8a88371220a4372
Finished request
```

Then they are retrieved during the *Access-Request*:
```
Received Access-Request Id 63 from 10.x.x.x:1645 to 10.x.x.x:1812 length 317
  User-Name = 'host/..............'
  Service-Type = Framed-User
  Cisco-AVPair = 'service-type=Framed'
  Framed-MTU = 1500
  Called-Station-Id = '......'
  Calling-Station-Id = '......'
  EAP-Message = 0x020d002b19001703010020faede9642fab863cc09665e68577e2f8b8cd31128186f9c4c930867f9b172906
  Message-Authenticator = 0xc2c6bac316b3caef9a26a3d1a193860c
  Cisco-AVPair = 'audit-session-id=0A02FC01000001631B2FD44B'
  NAS-Port-Type = Ethernet
  NAS-Port = 50135
  NAS-Port-Id = 'GigabitEthernet1/0/35'
  State = 0xe6d18ccaeddc95a5a8a88371220a4372
  NAS-IP-Address = 10.x.x.x
session-state: Found cached attributes
  MS-MPPE-Encryption-Policy += Encryption-Allowed
  MS-MPPE-Encryption-Types += RC4-40or128-bit-Allowed
  MS-MPPE-Send-Key += 0x41bec7335e750efe3e5d889041e406c5
  MS-MPPE-Recv-Key += 0xca6711a13fe9ebd182791f155b9d9e3f
  EAP-Message += 0x030c0004
  Message-Authenticator += 0x00000000000000000000000000000000
# Executing section authorize from file /etc/raddb/sites-enabled/default
```

And added to the *Access-Accept*:
```
Sent Access-Accept Id 63 from 10.x.x.x:1812 to 10.x.x.x:1645 length 292
  MS-MPPE-Recv-Key = 0x3e6c1050a5333f7a22bdbbd18b2ddad40e4b70fcdf0e889d8ebc77f523d32222
  MS-MPPE-Send-Key = 0xa711df8b3dec35b8b2fa35e875c3e85aeebb43d53c021f468ca57da6a61c167a
  EAP-Message = 0x030d0004
  Message-Authenticator = 0x00000000000000000000000000000000
  MS-MPPE-Encryption-Policy += Encryption-Allowed
  MS-MPPE-Encryption-Types += RC4-40or128-bit-Allowed
  MS-MPPE-Send-Key += 0x41bec7335e750efe3e5d889041e406c5
  MS-MPPE-Recv-Key += 0xca6711a13fe9ebd182791f155b9d9e3f
  EAP-Message += 0x030c0004
  Message-Authenticator += 0x00000000000000000000000000000000
Finished request
```

I'm not sure how to fix this correctly. Should these attributes be filtered in code or configuration? The `use_tunneled_reply` option filters them at `src/modules/rlm_eap/types/rlm_eap_peap/peap.c:447`.

**Comment by alandekok** [comment]
Edit the inner-tunnel virtual server, and add an additional "update" section after the following one:

	update {
		&outer.session-state: += &reply:
	}

	#
	#  These attributes are for the inner session only.
	#  They MUST NOT be sent in the outer reply.
	#
	update outer.session-state {
		MS-MPPE-Encryption-Policy !* ANY
		MS-MPPE-Encryption-Types !* ANY
		MS-MPPE-Send-Key !* ANY
		MS-MPPE-Recv-Key !* ANY
		Message-Authenticator !* ANY
		Proxy-State !* ANY
	}

That will filter out the attributes.

**Comment by johnnywalker** [comment]
Thanks for the quick answer!!

On Tue, Jan 27, 2015 at 9:59 AM, Alan DeKok <notifications@github.com>
wrote:

> Closed #893 <https://github.com/FreeRADIUS/freeradius-server/issues/893>
> via 570f1ad
> <https://github.com/FreeRADIUS/freeradius-server/commit/570f1ada1dedc5c5e20da07c0ce0f1b318ab994a>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeRADIUS/freeradius-server/issues/893#event-225490599>
> .
>



-- 
Johnny Walker
Senior Network Administrator
Designs for Health
860.752.7430 | 800.847.8302 x3017

-- 
NOTICE:  This E-mail (including attachments) is covered by the Electronic 
Communications Privacy Act,18 U.S.C. §§ 2510-2521, is confidential and 
legally privileged.  If you are not the intended recipient, you are hereby 
notified that any retention, dissemination, distribution, or copying of 
this communication is strictly prohibited.  Please reply to the sender that 
you have received the message in error; then delete it.  Thank you.

**Comment by johnnywalker** [comment]
Also need to filter out EAP-Message
